File: pipeline/register_summaries.js
// Phase 3c: after the summarize workflow, register v1 summaries and mark papers summarized.
// Usage: node --experimental-sqlite pipeline/register_summaries.js <topicId|all>
'use strict';
const fs = require('node:fs');
const path = require('node:path');
const { open, ROOT } = require('./lib/db');
const nowISO = () => new Date().toISOString();
const fileId = (id) => id.replace(/[^a-z0-9._-]/gi, '_').slice(0, 180);

function main() {
  const topicId = process.argv[2] || 'all';
  const db = open();
  const rows = topicId === 'all'
    ? db.prepare(`SELECT id, slug FROM papers WHERE status='pdf_downloaded'`).all()
    : db.prepare(`SELECT p.id, p.slug FROM papers p JOIN paper_topic pt ON pt.paper_id=p.id
                  WHERE pt.topic_id=? AND p.status='pdf_downloaded'`).all(topicId);

  let registered = 0, missing = 0;
  for (const { id, slug } of rows) {
    const vpath = path.join(ROOT, 'store', 'summaries', slug || fileId(id), 'v1.md');
    if (!fs.existsSync(vpath) || fs.statSync(vpath).size < 100) { missing++; continue; }
    const rel = path.relative(ROOT, vpath);
    db.prepare(
      `INSERT OR IGNORE INTO summary_versions (paper_id,version,path,based_on,note,created_at)
       VALUES (?,?,?,?,?,?)`
    ).run(id, 1, rel, '[]', '首次总结', nowISO());
    db.prepare(`UPDATE papers SET status='summarized', summarized_at=? WHERE id=?`).run(nowISO(), id);
    registered++;
  }
  db.close();
  console.log(`registered ${registered} v1 summaries; ${missing} still missing a summary file`);
}
main();

File: pipeline/summarize/register_summaries.py
"""After summarization, register v1 summaries and mark papers summarized.
Usage: python3 pipeline/summarize/register_summaries.py <topicId|all>
"""
import sys

# --- path shim: 让 `from lib...` 解析到 pipeline/lib，无论本文件在哪个子目录 ---
import os as _os, sys as _sys
_sys.path.insert(0, _os.path.dirname(_os.path.dirname(_os.path.abspath(__file__))))
from lib.db import open_db, ROOT, now_iso
from lib.slug import file_id


def main():
    topic_id = sys.argv[1] if len(sys.argv) > 1 else "all"
    conn = open_db()
    if topic_id == "all":
        rows = conn.execute("SELECT id, slug FROM papers WHERE status='pdf_downloaded'").fetchall()
    else:
        rows = conn.execute(
            """SELECT p.id, p.slug FROM papers p JOIN paper_topic pt ON pt.paper_id=p.id
                WHERE pt.topic_id=? AND p.status='pdf_downloaded'""", (topic_id,)).fetchall()

    registered, missing = 0, 0
    for r in rows:
        vpath = ROOT / "store" / "summaries" / (r["slug"] or file_id(r["id"])) / "v1.md"
        if not vpath.exists() or vpath.stat().st_size < 100:
            missing += 1
            continue
        rel = str(vpath.relative_to(ROOT))
        conn.execute(
            """INSERT OR IGNORE INTO summary_versions (paper_id,version,path,based_on,note,created_at)
               VALUES (?,?,?,?,?,?)""", (r["id"], 1, rel, "[]", "首次总结", now_iso()))
        conn.execute("UPDATE papers SET status='summarized', summarized_at=? WHERE id=?", (now_iso(), r["id"]))
        registered += 1
    conn.commit()
    conn.close()
    print(f"registered {registered} v1 summaries; {missing} still missing a summary file")


if __name__ == "__main__":
    main()
